src/ReaderThemeSupportFeatures.php

```diff
@@ -327,10 +403,20 @@ private function print_editor_font_sizes_styles( array $font_sizes ) {
 			if ( ! $this->has_required_feature_props( self::FEATURE_EDITOR_FONT_SIZES, $font_size ) ) {
 				continue;
 			}
+
+			// Just in case the font size is not in the expected format.
+			$font_size[ self::KEY_SIZE ] = $this->get_typography_value_and_unit( $font_size[ self::KEY_SIZE ] );
+
+			if ( empty( $font_size[ self::KEY_SIZE ] ) ) {
+				continue;
+			}
+
 			printf(
-				':root .is-%1$s-text, :root .has-%1$s-font-size { font-size: %2$spx; }',
+				':root .is-%1$s-text, :root .has-%1$s-font-size { font-size: %2$s; }',
 				sanitize_key( $font_size[ self::KEY_SLUG ] ),
-				(float) $font_size[ self::KEY_SIZE ]
+				function_exists( 'wp_get_typography_font_size_value' )
+					? esc_attr( wp_strip_all_tags( wp_get_typography_font_size_value( $font_size ) ) )
+					: esc_attr( wp_strip_all_tags( $font_size[ self::KEY_SIZE ] ) )
```

Comment:
This can just use `strip_tags()` since `esc_attr()`  could introduce entities which break CSS.

```suggestion
					? strip_tags( wp_get_typography_font_size_value( $font_size ) )
					: strip_tags( $font_size[ self::KEY_SIZE ] )
```